frontend/src/components/common/Resource/DeleteMultipleButton.tsx

```diff
@@ -0,0 +1,111 @@
+import _ from 'lodash';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { KubeObject } from '../../../lib/k8s/KubeObject';
+import { CallbackActionOptions, clusterAction } from '../../../redux/clusterActionSlice';
+import {
+  EventStatus,
+  HeadlampEventType,
+  useEventCallback,
+} from '../../../redux/headlampEventSlice';
+import { AppDispatch } from '../../../redux/stores/store';
+import ActionButton, { ButtonStyle } from '../ActionButton';
+import { ConfirmDialog } from '../Dialog';
+
+interface DeleteMultipleButtonProps {
+  items?: KubeObject[];
+  options?: CallbackActionOptions;
+  buttonStyle?: ButtonStyle;
+  afterConfirm?: () => void;
+}
+
+interface DeleteMultipleButtonDescriptionProps {
+  items?: KubeObject[];
+}
+
+function DeleteMultipleButtonDescription(props: DeleteMultipleButtonDescriptionProps) {
+  const { t } = useTranslation(['translation']);
+  return (
+    <p>
+      {t('Are you sure you want to delete following items?')}
+      <ul>
+        {props.items?.map(item => (
+          <li>{item.metadata.name}</li>
+        ))}
+      </ul>
+    </p>
+  );
+}
+
+export default function DeleteMultipleButton(props: DeleteMultipleButtonProps) {
+  const dispatch: AppDispatch = useDispatch();
+  const { items, options, afterConfirm, buttonStyle } = props;
+  const [openAlert, setOpenAlert] = React.useState(false);
+  const { t } = useTranslation(['translation']);
+  const dispatchDeleteEvent = useEventCallback(HeadlampEventType.DELETE_RESOURCES);
+
+  const deleteFunc = React.useCallback(
+    (items: KubeObject[]) => {
+      if (!items || items.length === 0) {
+        return;
+      }
+      const clonedItems = _.cloneDeep(items);
+      const itemsLength = clonedItems.length;
+
+      dispatch(
+        clusterAction(
+          () => {
+            clonedItems.forEach(item => {
+              item.delete();
+            });
```

Comment:
I wonder if we could run these in a promise and wait for all.

Reply:
I will check it out.

Reply:
What do you think?